### PR TITLE
Update tower-hedge to std::future

### PR DESCRIPTION
**Pull request by jonhoo**
Will rebase and mark as ready once #331 lands (which this depends on)

### CODE REVIEW

File: tower-hedge/src/latency.rs [hunk 1]
```diff
@@ -42,15 +49,15 @@ where
 impl<S, R, Request> Service<Request> for Latency<R, S>
 where
     S: Service<Request>,
-    S::Error: Into<super::Error>,
+    super::Error: From<S::Error>,
```
Comment (LucioFranco) [inline tower-hedge/src/latency.rs]: why this change?

File: tower-hedge/src/latency.rs [hunk 2]
```diff
@@ -62,24 +69,20 @@ where
...
 }
 
-impl<R, F> Future for ResponseFuture<R, F>
+impl<R, F, T, E> Future for ResponseFuture<R, F>
 where
     R: Record,
-    F: Future,
-    F::Error: Into<super::Error>,
+    F: Future<Output = Result<T, E>>,
+    super::Error: From<E>,
```
Comment (LucioFranco) [inline tower-hedge/src/latency.rs]: same here?

File: tower-hedge/src/rotating_histogram.rs [hunk 3]
```diff
@@ -1,8 +1,8 @@
 extern crate tokio_timer;
 
 use hdrhistogram::Histogram;
+use log::trace;
```
Comment (LucioFranco) [inline tower-hedge/src/rotating_histogram.rs]: tracing?

Comment (jonhoo) [reply tower-hedge/src/rotating_histogram.rs]: I want to change to `tracing` in a separate PR.

File: tower-hedge/src/latency.rs (hunk 1 above)
Comment (jonhoo) [reply tower-hedge/src/latency.rs]: Because for some reason, if I keep it as `Into<super::Error>`, I get a tonne of errors saying that the error does not implement `Send` and `Sync`. I'm not entirely sure why inverting the bound fixes the issue?

Comment (jonhoo) [reply tower-hedge/src/latency.rs]:
```
note: required because of the requirements on the impl of `std::convert::From<E>` for `std::boxed::Box<dyn std::error::Error + std::marker::Send + std::marker::Sync>`
```

Comment (jonhoo) [reply tower-hedge/src/latency.rs]: Actually, maybe @seanmonstar has some insight to provide here?

Comment (carllerche) [reply tower-hedge/src/latency.rs]: Can we just use `Into<Box<dyn Error>>` or whatever?

Comment (jonhoo) [reply tower-hedge/src/latency.rs]: That's what `super::Error` is:
https://github.com/tower-rs/tower/blob/e32a5ecd0cd24f5fabd139310ba854a8583ca2c4/tower-hedge/src/lib.rs#L50

Comment (seanmonstar) [reply tower-hedge/src/latency.rs]: You're seeing the error down in `Future::poll` because of the `?` operator, which uses `From` instead of `Into`.

Comment (jonhoo) [reply tower-hedge/src/latency.rs]: Thanks!
